core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -6414,6 +6414,14 @@ private Name() {} // prevent instantiation
   @ThreadSafe
   public enum Template {
     LOCALITY_TIER("alluxio.locality.%s", "alluxio\\.locality\\.(\\w+)"),
+    LOGICAL_MASTER_RPC_ADDRESS("alluxio.master.rpc.address.%s.%s",
+        "alluxio\\.master\\.rpc\\.address\\.([a-zA-Z_\\-0-9\\.]+).([a-zA-Z_\\-0-9\\.]+)"),
+    LOGICAL_MASTERS_NAME("alluxio.masters.%s",
+        "alluxio\\.masters\\.name\\.([a-zA-Z_\\-0-9\\.]+)"),
+    LOGICAL_ZOOKEEPER_NODES("alluxio.master.zookeeper.nodes.%s",
+        "alluxio\\.master\\.zookeeper\\.nodes\\.([a-zA-Z_\\-0-9\\.]+)"),
+    LOGICAL_ZOOKEEPER_ADDRESS("alluxio.master.zookeeper.address.%s.%s",
+        "alluxio\\.master\\.zookeeper\\.address\\.([a-zA-Z_\\-0-9\\.]+)\\.([a-zA-Z_\\-0-9\\.]+)"),
```

Comment:
Can you separate out the regex pattern strings into named variables for clarity. eg:

```
String LOGICAL_NAME_PATTERN_STRING="([a-zA-Z_\\-0-9\\.]+)";
...
public enum Template {
  LOGICAL_MASTER_RPC_ADDRESS("alluxio.master.rpc.address.%s.%s",
        String.format("alluxio\\.master\\.rpc\\.address\\.%s\\.%s", LOGICAL_NAME_PATTERN_STRING, LOGICAL_NAME_PATTERN_STRING),
  ...
}
```

---
Also `alluxio.masters` is not an existing property prefix from what I can tell from [our documentation](https://docs.alluxio.io/os/user/stable/en/reference/Properties-List.html).

I apologize for not catching that in your earlier comment. @ggezer wdyt about refactoring it to be `alluxio.master.logical.[cluster-name]` and `alluxio.master.logical.zookeeper.[cluster-name]`? I also personally would prefer `nameservices` over `logical` to clearly indicate that this is akin to HDFS nameservices.